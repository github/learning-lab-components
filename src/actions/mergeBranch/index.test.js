const mergeBranch = require('./index')
const mockContext = require('../../tests/mockContext')

describe('mergeBranch', () => {
  let context

  beforeEach(() => {
    context = mockContext({}, {
      repos: {
        merge: jest.fn(() => Promise.resolve({ data: { number: 1 } }))
      }
    })
  })

  it('merges a branch into master', async () => {
    await mergeBranch(context, { head: 'branch' })
    expect(context.github.repos.merge).toHaveBeenCalled()
    expect(context.github.repos.merge.mock.calls).toMatchSnapshot()
  })

  it('merges a branch into the provided base', async () => {
    await mergeBranch(context, { head: 'branch', base: 'dev' })
    expect(context.github.repos.merge).toHaveBeenCalled()
    expect(context.github.repos.merge.mock.calls).toMatchSnapshot()
  })
})
