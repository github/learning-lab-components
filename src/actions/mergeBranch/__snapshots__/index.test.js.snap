// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mergeBranch merges a branch into master 1`] = `
Array [
  Array [
    Object {
      "base": "master",
      "head": "branch",
      "owner": "JasonEtco",
      "repo": "example",
    },
  ],
]
`;

exports[`mergeBranch merges a branch into the provided base 1`] = `
Array [
  Array [
    Object {
      "base": "dev",
      "head": "branch",
      "owner": "JasonEtco",
      "repo": "example",
    },
  ],
]
`;
